### PR TITLE
Code Cleanup: remove forkchoicer from beacon node

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

Cleanup

**What does this PR do? Why is it needed?**

Forkchoicer is only used in the beaconchain service and can be cleaned up/removed from the node.

### CODE REVIEW

File: beacon-chain/node/node.go [hunk 1]
```diff
@@ -609,13 +596,19 @@ func (b *BeaconNode) registerBlockchainService() error {
...
-		blockchain.WithForkChoiceStore(b.forkChoiceStore),
 		blockchain.WithAttestationService(attService),
 		blockchain.WithStateGen(b.stateGen),
 		blockchain.WithSlasherAttestationsFeed(b.slasherAttestationsFeed),
 		blockchain.WithFinalizedStateAtStartUp(b.finalizedStateAtStartUp),
 		blockchain.WithProposerIdsCache(b.proposerIdsCache),
 	)
+
+	if !features.Get().DisableForkchoiceDoublyLinkedTree {
+		opts = append(opts, blockchain.WithForkChoiceStore(doublylinkedtree.New()))
```
Comment (potuz) [inline beacon-chain/node/node.go:607]: Perhaps avoid the double negation here?
```suggestion
	if features.Get().DisableForkchoiceDoublyLinkedTree {
		opts = append(opts, blockchain.WithForkChoiceStore(protoarray.New()))
```
But anyway I think it's cleaner to move the forkchoice structure away from `blockchain.cfg` and into the `blockchain.Service` structure. And initialize it from blockchain instead of here. I don't feel strongly about this though.

File: beacon-chain/node/node.go [hunk 2]
```diff
@@ -843,7 +836,7 @@ func (b *BeaconNode) registerRPCService() error {
 	return b.services.RegisterService(rpcService)
 }
 
-func (b *BeaconNode) registerPrometheusService(cliCtx *cli.Context) error {
+func (b *BeaconNode) registerPrometheusService(_ *cli.Context) error {
```
Comment (potuz) [inline beacon-chain/node/node.go:839]: why this change?

Comment (james-prysm) [reply beacon-chain/node/node.go:839]: cliCtx isn't used in this so usually deep source complains when you touch the file and there's unused arguments

File: beacon-chain/node/node.go (hunk 1 above)
Comment (james-prysm) [reply beacon-chain/node/node.go]: I think this clearly shows this dependency, could we leave it in cfg for now? I can revisit it later too

**Review by rauljordan — APPROVED** [review]
